### PR TITLE
ADDED: singlevideo, videolisting

### PR DESCRIPTION
**Pull request by Adarsh-Patel-Dev**
Files to check: 
 Videolisting
 SingleVideo
 
 
image

image

### CODE REVIEW

File: src/components/Card/SinglevideoCard/SingleVideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      </div> 
+
+       <div className="card--body--single-video">
+
+                <div className="card--details--singlevideo">
+                    <h4 className="card--title">{title}</h4>
+                    <div className="card-views">{views} . {old}</div>
+                    <div className="actions-buttons">
+                      <span><AiOutlineLike/>10k</span>
+                      <span><AiOutlineDislike/></span>
```
Comment (yaswanthmyneni) [inline src/components/Card/SinglevideoCard/SingleVideoCard.jsx]: changes: I think the span tags are not required here.

File: src/components/Card/SinglevideoCard/singlevideo.css [hunk 2]
```diff
@@ -0,0 +1,72 @@
+
+.card--single-video {
+    width: 67.6%;
+    height: auto;
+    background-color: var(--WHITE);
+    color: var(--BLACK-80);
+    border-radius: 8px;
+    padding: 20px;
```
Comment (yaswanthmyneni) [inline src/components/Card/SinglevideoCard/singlevideo.css]: changes: use rem for more than 16px

File: src/components/Card/card.css [hunk 3]
```diff
@@ -113,8 +124,8 @@
 }
 
 .card-channel{
-    color: rgb(69, 69, 69);
-    font-weight: 400;
+    color: black;
```
Comment (yaswanthmyneni) [inline src/components/Card/card.css]: changes: use css variables for colors

File: src/context/liked-video-context.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        const response = await axios.get({
+            method : 'GET',
+            url: '/api/user/likes',
+            headers: { authorization: localStorage.getItem('token')},
+
+        })
+
+        if(response.status === 200){
+            setLikeVideo(response.data.likes)
+        }
```
Comment (yaswanthmyneni) [inline src/context/liked-video-context.js]: suggestion: use try catch for error handling

File: src/context/video-listing-context.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    async function getVideoData() {
+        try {
+            const response = await axios.get('/api/videos');
+            if(response.status === 200){
+                setVideolist(response.data.videos)
+                console.log('res from videolisting',response.data.videos)
+            }
+        } catch (error){
+            console.log(error);
```
Comment (yaswanthmyneni) [inline src/context/video-listing-context.js]: suggestion: use console.error for displaying errors

File: src/context/video-listing-context.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    
+    const [videoList, setVideolist] = useState([]);
+    const [category, setCategory] = useState([]);
+
+    async function getVideoData() {
+        try {
+            const response = await axios.get('/api/videos');
+            if(response.status === 200){
+                setVideolist(response.data.videos)
+                console.log('res from videolisting',response.data.videos)
```
Comment (yaswanthmyneni) [inline src/context/video-listing-context.js]: changes: remove console.log before pushing to github everywhere

File: src/index.js [hunk 7]
```diff
@@ -2,14 +2,20 @@ import React from "react";
...
 import { makeServer } from "./server";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+  <VideoListingProvider>
+    <BrowserRouter>
```
Comment (yaswanthmyneni) [inline src/index.js:18]: changes: place browserRouter at the top of all custom ContextProviders

File: src/pages/video-listing/VideoListing.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        <div className="card--container">
+           
+           {
+               videoList?.map((video)=> (
+                   <CardHorizontal 
+                   key ={video._id}
+                   video ={ video }
+
+                   />
+
```
Comment (yaswanthmyneni) [inline src/pages/video-listing/VideoListing.jsx]: changes: remove extra spaces here

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes

File: src/pages/video-listing/VideoListing.jsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  Navigation,
+  CardHorizontal
+  } from '../../components/';
+import axios from 'axios';
+
+
+function VideoListing() {
+
+    const { videoList } = useVideoListing();
+    console.log(videoList, 'this is from videolisting')
```
Comment (anasm11) [inline src/pages/video-listing/VideoListing.jsx]: you can remove console logs before committing.

File: src/components/Card/SinglevideoCard/SingleVideoCard.jsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+import './singlevideo.css'
+
+function SingleVideoCard() {
+
+  const {videoId} = useParams();
+  console.log(videoId);
+
+  const { videoList } = useVideoListing();
+
+  console.log( videoList , 'this is from single videopage')
```
Comment (anasm11) [inline src/components/Card/SinglevideoCard/SingleVideoCard.jsx]: same here

**Review by anasm11 — COMMENTED** [review]
Didnt check the preview as link was not provided. The code looks good.

File: src/components/Card/SinglevideoCard/SingleVideoCard.jsx (hunk 1 above)
Comment (Adarsh-Patel-Dev) [reply src/components/Card/SinglevideoCard/SingleVideoCard.jsx]: No i have styled icons using `actions-buttons span`

**Comment by Adarsh-Patel-Dev** [comment]
Completed:
videolisting
single video
all context

**Pull request merged**